src/runtime/utils/entry.mjs

```diff
@@ -56,3 +56,57 @@ export function initReducedView() {
     tmp.remove();
   });
 }
+
+export async function hasBatteryPerformanceIssue(videoBlob) {
+  await isBatteryLow();
+  await canVideoPlay(videoBlob);
+}
+
+/**
+ * Checks if battery still has enough energy.
+ * This check is for Chrome and all other browsers that support this setting.
+ *
+ * Condition is: The device is not charging and Battery is below <= 20%.
+ * @see https://blog.google/products/chrome/new-chrome-features-to-save-battery-and-make-browsing-smoother/
+ * @see https://developer.chrome.com/blog/memory-and-energy-saver-mode/
+ **/
+async function isBatteryLow() {
+  const MIN_BATTERY_LEVEL = 0.2;
+
+  try {
+    const battery = await window.navigator.getBattery();
+    if (!battery.charging && battery.level <= MIN_BATTERY_LEVEL) {
+      throw new Error('Battery is low.');
+    }
+  } catch (error) {
+    // Ignore Check
+  }
+}
+
+/**
+ * Checking whether a video can be played.
+ * This check is for IOS and checks if the power saving mode is enabled.
+ *
+ * In this case no video will be played automatically.
+ */
+export async function canVideoPlay(blob) {
+  const VIDEO_TIMEOUT = 250;
+
+  const video = document.createElement('video');
+  video.muted = true;
```

Comment:
do we need that. it seems to be a duplicate -> next line

Reply:
All `setAttribute` have been removed and replaced with set properties.